publisher/src/components/MetricConfiguration/ContextConfiguration.tsx

```
@@ -32,189 +33,185 @@ import {
-                      })
-                    }
+export const ContextConfiguration: React.FC<MetricContextConfigurationProps> =
+  observer(({ saveUpdatedMetricSettings }) => {
+    const { metricConfigStore } = useStore();
+    const {
+      activeMetricKey,
+      activeSystem,
+      contexts,
+      updateContextValue,
+      getActiveSystemMetricKey,
+    } = metricConfigStore;
+
+    const systemMetricKey = getActiveSystemMetricKey();
+    const activeContextKeys =
+      (contexts[systemMetricKey] && Object.keys(contexts[systemMetricKey])) ||
+      [];
+
+    const debouncedSave = useRef(
+      debounce(saveUpdatedMetricSettings, 1500)
+    ).current;
+
+    return (
+      <MetricContextContainer>
+        <MetricContextHeader>Context</MetricContextHeader>
+        <Subheader>
+          Anything entered here will appear as the default value for all
+          reports. If you are entering data for a particular month, you can
+          still replace this as necessary.
+        </Subheader>
+
+        {activeContextKeys.map((contextKey) => {
+          const currentContext = contexts[systemMetricKey][contextKey];
+
+          return (
+            <MetricContextItem key={contextKey}>
+              {/* Binary Context Questions */}
+              {currentContext.type === "BOOLEAN" && (
+                <>
+                  <Label noBottomMargin>{currentContext.display_name}</Label>
+                  <RadioButtonGroupWrapper>
+                    <BinaryRadioButton
+                      type="radio"
+                      id={`${contextKey}-yes`}
+                      name={contextKey}
+                      label="Yes"
+                      value="yes"
+                      checked={currentContext.value === "yes"}
+                      onChange={() => {
+                        const updatedSetting = updateContextValue(
+                          activeSystem as string,
+                          activeMetricKey as string,
```

can this be undefined?

---

Maybe the function can just throw an error or log a warning if undefined?

---

Yeah - great call - this is tricky. Technically they can, but things are initialized by the time it gets to this point. 

Let me think about this - if we update the function to throw an error if undefined, then we'll need an extra step of checking if `updatedSetting` is undefined before saving, right?

---

Worked it out in this commit: [ed985e5](https://github.com/Recidiviz/justice-counts/pull/132/commits/ed985e5bd31081d7c07409f72567d7cd4af03033)

Thank you!